### PR TITLE
feat(IMS): ims support fix min ram, max ram and description

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:
  ims support fix min ram, max ram and description
**Release note**:

```release-note
  ims support fix min ram, max ram and description
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/ims/' TESTARGS='-run TestAccImsImage_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ims/ -v -run TestAccImsImage_ -timeout 360m -parallel 4
=== RUN   TestAccImsImage_basic
=== PAUSE TestAccImsImage_basic
=== RUN   TestAccImsImage_withEpsId
=== PAUSE TestAccImsImage_withEpsId
=== RUN   TestAccImsImage_wholeImage_withServer
=== PAUSE TestAccImsImage_wholeImage_withServer
=== RUN   TestAccImsImage_wholeImage_withBackup
=== PAUSE TestAccImsImage_wholeImage_withBackup
=== CONT  TestAccImsImage_basic
=== CONT  TestAccImsImage_wholeImage_withServer
=== CONT  TestAccImsImage_withEpsId
=== CONT  TestAccImsImage_wholeImage_withBackup
--- PASS: TestAccImsImage_wholeImage_withBackup (59.97s)
--- PASS: TestAccImsImage_withEpsId (401.36s)
--- PASS: TestAccImsImage_basic (532.55s)
--- PASS: TestAccImsImage_wholeImage_withServer (1104.58s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ims       1104.633s

make testacc TEST='./huaweicloud/services/acceptance/ims/' TESTARGS='-run TestAccImsImageCopy_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ims/ -v -run TestAccImsImageCopy_ -timeout 360m -parallel 4
=== RUN   TestAccImsImageCopy_basic
=== PAUSE TestAccImsImageCopy_basic
=== RUN   TestAccImsImageCopy_basic_cross_region
=== PAUSE TestAccImsImageCopy_basic_cross_region
=== CONT  TestAccImsImageCopy_basic
=== CONT  TestAccImsImageCopy_basic_cross_region
--- PASS: TestAccImsImageCopy_basic (621.60s)
--- PASS: TestAccImsImageCopy_basic_cross_region (793.13s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ims       793.178s
```

### CODE REVIEW

File: huaweicloud/services/ims/resource_huaweicloud_images_image.go [hunk 1]
```diff
@@ -45,17 +48,17 @@ func ResourceImsImage() *schema.Resource {
...
 				Type:     schema.TypeInt,
 				Optional: true,
-				ForceNew: true,
+				Computed: true,
 			},
 			"min_ram": {
 				Type:     schema.TypeInt,
 				Optional: true,
-				ForceNew: true,
+				Computed: true,
```
Comment (Lance52259) [inline huaweicloud/services/ims/resource_huaweicloud_images_image.go]: Does these parameters not support updating to the null value?

File: docs/resources/images_image_copy.md [hunk 2]
```diff
@@ -71,6 +68,13 @@ The following arguments are supported:
   and it is mandatory if you are replicating a full-ECS image.
   Changing this parameter will create a new resource.
 
+* `min_ram` - (Optional, Int) The minimum memory of the image in the unit of MB. The default value is 0,
+  indicating that the memory is not restricted.
+
+* `max_ram` - (Optional, Int) The maximum memory of the image in the unit of MB.
+
+* `description` - (Optional, String) A description of the image.
```
Comment (deer-hang) [inline docs/resources/images_image_copy.md]: r/image/copy image.
Please fix the other parts.

File: huaweicloud/services/ims/resource_huaweicloud_images_image.go [hunk 3]
```diff
@@ -45,17 +48,17 @@ func ResourceImsImage() *schema.Resource {
 			"description": {
 				Type:     schema.TypeString,
 				Optional: true,
-				ForceNew: true,
+				Computed: true,
```
Comment (deer-hang) [inline huaweicloud/services/ims/resource_huaweicloud_images_image.go]: move down the computed field

File: huaweicloud/services/ims/resource_huaweicloud_images_image.go [hunk 4]
```diff
@@ -335,13 +338,17 @@ func resourceImsImageRead(_ context.Context, d *schema.ResourceData, meta interf
...
 		d.Set("description", img.Description),
+		d.Set("min_ram", img.MinRam),
 		d.Set("visibility", img.Visibility),
 		d.Set("disk_format", img.DiskFormat),
 		d.Set("image_size", img.ImageSize),
 		d.Set("enterprise_project_id", img.EnterpriseProjectID),
 		d.Set("checksum", img.Checksum),
 		d.Set("status", img.Status),
 	)
+	if maxRAM, err := strconv.Atoi(img.MaxRam); err == nil {
```
Comment (deer-hang) [inline huaweicloud/services/ims/resource_huaweicloud_images_image.go:351]: Leave an  error message here, if strconv faild.

File: huaweicloud/services/ims/resource_huaweicloud_images_image.go [hunk 5]
```diff
@@ -432,18 +439,54 @@ func resourceImsImageUpdate(ctx context.Context, d *schema.ResourceData, meta in
...
+			Value: d.Get("min_ram").(int),
+		}
+		maxRAM := cloudimages.UpdateImageProperty{
+			Op:    cloudimages.ReplaceOp,
+			Name:  "max_ram",
+			Value: strconv.Itoa(d.Get("max_ram").(int)),
+		}
+		updateOpts = append(updateOpts, name)
+		updateOpts = append(updateOpts, minRAM)
+		updateOpts = append(updateOpts, maxRAM)
```
Comment (deer-hang) [inline huaweicloud/services/ims/resource_huaweicloud_images_image.go]: How about make the append together in aline

File: huaweicloud/services/ims/resource_huaweicloud_images_image.go [hunk 6]
```diff
@@ -432,18 +439,54 @@ func resourceImsImageUpdate(ctx context.Context, d *schema.ResourceData, meta in
...
+			Op:    cloudimages.ReplaceOp,
+			Name:  "__description",
+			Value: d.Get("description").(string),
+		}
+		updateOpts = append(updateOpts, description)
+
+		log.Printf("[DEBUG] Update description Options: %#v", updateOpts)
+		_, err = cloudimages.Update(imsClient, d.Id(), updateOpts).Extract()
+		if err != nil {
+			err = dealModifyDescriptionErr(d, imsClient, err)
```
Comment (deer-hang) [inline huaweicloud/services/ims/resource_huaweicloud_images_image.go:484]: It is better to add a description for this  function

File: huaweicloud/services/ims/resource_huaweicloud_images_image_copy.go [hunk 7]
```diff
@@ -218,18 +248,54 @@ func resourceImsImageCopyUpdate(ctx context.Context, d *schema.ResourceData, met
...
+			Value: d.Get("min_ram").(int),
+		}
+		maxRAM := cloudimages.UpdateImageProperty{
+			Op:    cloudimages.ReplaceOp,
+			Name:  "max_ram",
+			Value: strconv.Itoa(d.Get("max_ram").(int)),
+		}
+		updateOpts = append(updateOpts, name)
+		updateOpts = append(updateOpts, minRAM)
+		updateOpts = append(updateOpts, maxRAM)
```
Comment (deer-hang) [inline huaweicloud/services/ims/resource_huaweicloud_images_image_copy.go]: ditto

**Pull request merged**